### PR TITLE
redirect: initial test for Redirect

### PR DESCRIPTION
**Pull request by peterhellberg**
A small test case for redirect based on the rewrite test.
Testing both that the Location header and response body is set to the expected values.

Should we skip verification of the body since it is generated in `http.Redirect`?

### CODE REVIEW

**Comment by mholt** [comment]
Yeah, I don't think we need to check the body - at least, not like this. It might be more appropriate to check if the dummy handler was called (which it shouldn't be when performing a redirect). One way to do this would be to make sure the body does NOT equal what the dummy handler writes.

Or, another way: you could inline the dummy handler function and create a variable in the test function, make sure the dummy handler sets it to `true` or something, then check after a redirect to make sure the value is *not* `true`.

In other words, part of this is testing that the Next handler was or wasn't called.

Otherwise, this looks good!

**Comment by mholt** [comment]
Beautiful - thanks!
